Add an emeritus role to governance (#105)
containerd/project: GOVERNANCE.md

After you've informed other maintainers, create a pull request to remove
yourself from the MAINTAINERS file.

## Handling inactive maintainers

Similar to the procedure for adding new maintainers, existing maintainers can
be removed from the list if they do not show consistent activity on the
project. Depending on the existing role, committer or reviewer, there are

> variations to the removal process detailed as follows.

Reviewer: ```suggestion
variations to the inactivity process detailed as follows.
```

Author: Thanks, fixed this in the final push

### Committer inactivity process

For committers, the maintainers periodically review committer project
activity. The maintainers use the CNCF "devstats" community metrics tool for
data on user activity statistics.

If a committer has shown no activity over a six-month period, the committer
will be contacted to notify them of their activity status and offer a move to
an Emeritus role. There is no automatic change of status in the project. First,
the activity status is discussed directly with the committer, and second,
maintainers discuss whether other non-tracked contributions to the project
reflect an ongoing, active participation in the project.

Based on the outcome of these discussions, the committer may choose to remove
themselves from the MAINTAINERS file, as noted above in the "Stepping down
policy" section. If another maintainer opens a pull request to the MAINTAINERS

> file on their behalf for removal, the PR must be approved by 2/3 of the current
> committers. The voting and discussion period is seven days, after which a

Reviewer 2: ```suggestion
file on their behalf for removal, the PR must be approved by 2/3 of the current
committers or by the committer who is being removed. The voting and discussion period is seven days, after which a
```

It'd be useful to allow a short-circuit here even if the inactive maintainer isn't the one opening the PR themselves.

Reviewer 3: @estesp PTAL

Author: ✅ updated!

committer will verify the votes, merge the pull request, and apply the changes
to the organization.

### Reviewer inactivity process

While reviewers are not voting members of the containerd project, the maintainers
may assess reviewer activity periodically and recommend removals for any
completely inactive reviewers.

Reviewers may remove themselves from the MAINTAINERS file if they wish to or
any existing maintainer may open a pull request to the MAINTAINERS file on their

> behalf. If the pull request is not initiated by the reviewer, it must be approved
> by 2/3 of the current committers. The voting and discussion period is seven days,

Reviewer 2: ```suggestion
behalf. If the pull request is not initiated by the reviewer, it must be approved
by 2/3 of the current committers or by the reviewer who is being removed. The voting and discussion period is seven days,
```

Same comment about short-circuit here.

Author: ✅ updated!

## Removal of inactive maintainers

Similar to the procedure for adding new maintainers, existing maintainers can
be removed from the list if they do not show consistent activity on the
project. Depending on the existing role, committer or reviewer, there are
variations to the removal process detailed as follows.

> ### Committer removal

Reviewer: ```suggestion
### Committer inactivity process 
```

Author: ✅

For committers, the maintainers periodically review committer project
activity. The maintainers use the CNCF "devstats" community metrics tool for
data on user activity statistics.

> If a committer has shown no activity over a six-month period, the committer

Reviewer: ```suggestion
If a committer has shown no activity over a six-month period, a process will begin to review the committer's status as an active maintainer. The committer
```

Author: Since you had 2 suggestions for the same sentence I took the latter one (below) as I believe the "process" part is detailed in the sentences following.

> will be contacted to notify them of their proposed Emeritus role. There is

Reviewer: ```suggestion
will be contacted to notify them of their status and the option of moving to an Emeritus role. There is
```

Author: Done

no automatic change of status in the project. First, the activity status is
discussed directly with the committer, and second, maintainers discuss whether
other non-tracked contributions to the project reflect an ongoing, active
participation in the project.

Based on the outcome of these discussions, the committer may choose to remove
themselves from the MAINTAINERS file, or another maintainer may open a
pull request to the MAINTAINERS file on their behalf for removal.

> Removals must be approved by 2/3 of the current committers. The voting

Reviewer 2: ```suggestion
Removals must be approved by either 2/3 of the current committers or by the committer being removed plus one other current committer. The voting
```

Reviewer 2: It looks like maybe this is covered by the `## Stepping down policy` above.  Or we could clarify that part and explicitly cover the approval requirements in that section and use this section only for removal without stepping down.

Reviewer 3: > by the committer being removed plus one other current committer

`plus one` might not be needed, as anyone should be allowed to step down on their own will.

Reviewer 2: Agreed.  Someone needs to apply the changes to the org though (and that can't be the person stepping down).

> Removals must be approved by 2/3 of the current committers. The voting

Reviewer 3: ```suggestion
Removals must be approved by 2/3 of the current committers who voted within 14 days. The voting
```

To avoid potential dead-lock when more than 1/3 of the committers are inactive.

Reviewer 4: I don't think we need to change our threshold voting rules at this time. We have had some voting go right up to the 7 days, but even with 1/3 inactive right now, we have always gotten the votes. Even inactive committers are usually willing to come back on important votes, but the idea here is that in the future we will always have a comfortable 2/3 actively participating in voting an discussions.

We can continue to discuss evolving our governance model to better recognize both contributor activity as well as the wider diversity of containerd projects.

and discussion period is seven days, after which a committer will verify the votes,
merge the pull request, and apply the changes to the organization.

> ### Emeritus maintainers
> 
> For committers who are stepping down, the project would like to memorialize their

Reviewer 2: ```suggestion
## Emeritus maintainers

For committers who are stepping down or are removed for being inactive, the project would like to memorialize their
```

Let's move this up a level (in heading) so it covers both `## Stepping down policy` and `## Removal of inactive maintainers`.

Author: ✅ reworked the ordering of the sections and moved up a level

contributions to the project by recognizing them as Emeritus maintainers in
the EMERITUS.md file. The EMERITUS.md file will include a brief paragraph
summarizing their contribution to the containerd project and recognize them as
permanent Emeritus members of the community.

> If in the future an Emeritus maintainer has the desire or ability to return to

Reviewer 5: I wonder if this needs to require any demonstration of new activity in the project?

Reviewer 2: I think an emeritus committer volunteering to become more active is sufficient; if they're unable to follow-through with that we can handle that through discussion.

Reviewer 6: agree with Samuel, need to make this as easy/painless as possible

contributing to the project, Emeritus maintainers can submit a pull request
reversing their removal from the MAINTAINERS file and approval only requires

> 2 LGTMs from existing committers to return to full committer status in the

Reviewer 2: ```suggestion
2 LGTMs from current committers to return to full committer status in the
```

Author: ✅

project.

> ### Reviewer removal

Reviewer: ```suggestion
### Reviewer inactivity process
```

Author: ✅

While reviewers are not voting members of the containerd project, the maintainers
may assess reviewer activity periodically and recommend removals for any
completely inactive reviewers.

Reviewers may remove themselves from the MAINTAINERS file if they wish to or
any existing maintainer may open a pull request to the MAINTAINERS file on their

> behalf. Removals must be approved by 2/3 of the current committers. The voting

Reviewer 6: Does this merit a lower bar?

Reviewer 2: Adding reviewers only requires 1/3.  Maybe we move this to a simple majority and clarify it's not needed for stepping down?

```suggestion
behalf. Non-voluntary removals must be approved by a simple majority of the current committers. The voting
```

Reviewer: I think after we've moved the existing ex-maintainers who "stepped down" to reviewer, then yes perhaps 1/3 or simple majority..  Suggest leaving it at 2/3 for now for that reason.
